seed_reader/lib/routes/generate_seed/generate_seed_page.dart

```diff
@@ -0,0 +1,171 @@
+import 'dart:convert';
+
+import 'package:flutter/material.dart';
+import 'package:provider/provider.dart';
+import 'package:qr_flutter/qr_flutter.dart';
+import '../../../generated/l10n.dart';
+import '../../components/countdown_timer.dart';
+import '../../design_tokens/dimensions.dart';
+import '../../errors.dart';
+import '../../models/seed.dart';
+import '../../routes/generate_seed/generate_seed_page_bloc.dart';
+
+class GenerateSeedPage extends StatelessWidget {
+  const GenerateSeedPage({Key? key}) : super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      appBar: AppBar(
+        title: Text(S.of(context).generateSeedPageTitle),
+      ),
+      body: Consumer<GenerateSeedPageBloc>(
+        builder: (_, GenerateSeedPageBloc bloc, __) {
+          return StreamBuilder<Seed?>(
+            stream: bloc.observeSeed(),
+            builder: (_, AsyncSnapshot<Seed?> snapshot) {
+              if (snapshot.hasData) {
+                return _SuccessStateView(seed: snapshot.data!, bloc: bloc);
+              }
+              if (snapshot.hasError) {
+                return _ErrorStateView(
+                  bloc: bloc,
+                  error: snapshot.error,
+                );
+              }
+              return const _LoadingStateView();
+            },
+          );
```

Comment:
suggestion: I would think about renaming these to exclude state since our `snapshot` is our state. `ErrorView` would be sufficient for example I think.